test/test_archive.py

```diff
@@ -0,0 +1,1226 @@
+):
+    # Creates a folder where to save the zip
+    archives_folder = f"{working_dir}/my_archives/"
+    Path(archives_folder).mkdir()
+
+    result = run_command(
+        f'archive "{working_dir}/sketch_simple" "{archives_folder}/my_custom_sketch" --include-build-dir', copy_sketch
+    )
+    assert result.ok
+
+    archive = zipfile.ZipFile(f"{archives_folder}/my_custom_sketch.zip")
+    archive_files = archive.namelist()
+
+    assert "sketch_simple/doc.txt" in archive_files
+    assert "sketch_simple/header.h" in archive_files
+    assert "sketch_simple/merged_sketch.txt" in archive_files
+    assert "sketch_simple/old.pde" in archive_files
+    assert "sketch_simple/other.ino" in archive_files
+    assert "sketch_simple/s_file.S" in archive_files
+    assert "sketch_simple/sketch_simple.ino" in archive_files
+    assert "sketch_simple/src/helper.h" in archive_files
+    assert "sketch_simple/build/adafruit.samd.adafruit_feather_m0/sketch_simple.ino.hex" in archive_files
+    assert "sketch_simple/build/adafruit.samd.adafruit_feather_m0/sketch_simple.ino.map" in archive_files
+    assert "sketch_simple/build/arduino.avr.uno/sketch_simple.ino.eep" in archive_files
+    assert "sketch_simple/build/arduino.avr.uno/sketch_simple.ino.hex" in archive_files
+    assert "sketch_simple/build/arduino.avr.uno/sketch_simple.ino.with_bootloader.hex" in archive_files
+
+    archive.close()
+
+
+def test_archive_absolute_sketch_path_with_absolute_zip_path_and_name_with_extension_with_include_build_dir_flag(
+    run_command, copy_sketch, working_dir
+):
+    # Creates a folder where to save the zip
+    archives_folder = f"{working_dir}/my_archives/"
+    Path(archives_folder).mkdir()
+
+    result = run_command(
+        f'archive "{working_dir}/sketch_simple" "{archives_folder}/my_custom_sketch.zip" --include-build-dir',
+        copy_sketch,
+    )
+    assert result.ok
+
+    archive = zipfile.ZipFile(f"{archives_folder}/my_custom_sketch.zip")
+    archive_files = archive.namelist()
+
+    assert "sketch_simple/doc.txt" in archive_files
+    assert "sketch_simple/header.h" in archive_files
+    assert "sketch_simple/merged_sketch.txt" in archive_files
+    assert "sketch_simple/old.pde" in archive_files
+    assert "sketch_simple/other.ino" in archive_files
+    assert "sketch_simple/s_file.S" in archive_files
+    assert "sketch_simple/sketch_simple.ino" in archive_files
+    assert "sketch_simple/src/helper.h" in archive_files
+    assert "sketch_simple/build/adafruit.samd.adafruit_feather_m0/sketch_simple.ino.hex" in archive_files
+    assert "sketch_simple/build/adafruit.samd.adafruit_feather_m0/sketch_simple.ino.map" in archive_files
+    assert "sketch_simple/build/arduino.avr.uno/sketch_simple.ino.eep" in archive_files
+    assert "sketch_simple/build/arduino.avr.uno/sketch_simple.ino.hex" in archive_files
+    assert "sketch_simple/build/arduino.avr.uno/sketch_simple.ino.with_bootloader.hex" in archive_files
```

Comment:
This is always the same, maybe it's more convenient to put these checks in a function?

Reply:
Yep, will do.